c++/src/capnp/rpc-test.c++

```diff
@@ -1303,6 +1311,46 @@ TEST(Rpc, Abort) {
   EXPECT_TRUE(conn->receiveIncomingMessage().wait(context.waitScope) == nullptr);
 }
 
+KJ_TEST("handles exceptions thrown during disconnect") {
+  // This is similar to the earlier "abort" test, but throws an exception on
```

Comment:
Not sure if there might be a better way to test this condition... I just happened to find that the "abort" test case was one that exercised the exception handler I was interested in.